libminifi/src/utils/FileReaderCallback.cpp

```diff
@@ -0,0 +1,72 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "utils/FileReaderCallback.h"
+
+#include <cinttypes>
+
+#include "Exception.h"
+#include "core/logging/LoggerConfiguration.h"
+
+namespace {
+
+constexpr std::size_t BUFFER_SIZE = 4096;
+
+}  // namespace
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+
+FileReaderCallback::FileReaderCallback(const std::string& file_name)
+    : logger_(core::logging::LoggerFactory<FileReaderCallback>::getLogger()) {
+  logger_->log_debug("Opening %s", file_name);
+  input_stream_.open(file_name.c_str(), std::fstream::in | std::fstream::binary);
+  if (!input_stream_.is_open() || !input_stream_.good()) {
```

Comment:
I copied this mindlessly from `ProcessSession`, but I think you're right, `is_open()` and `good()` are the same immediately after `open()`, so we were checking it twice.  I have removed the `good()` check, because I think `is_open()` reads better in combination with `open()`.